chapters/annotations.tex

```diff
@@ -1119,7 +1119,7 @@ \section{Annotations for the Graphical User Interface}\label{annotations-for-the
 
 \begin{lstlisting}[language=grammar]
   preferred-view-annotation:
-     annotation "(" preferredView "=" ("info" | "diagram" | "text") ")"
+     annotation "(" preferredView "=" ("""info""" | """diagram""" | """text""") ")"
```

Comment:
As indicated by the comments in #2988, this could actually be considered a backwards incompatible change.  Therefore, I think we need a language group decision before just throwing out the non-string variant.

Also, I'm not convinced that triple double quotes is the proper way to quote a string literal in the grammar.

Reply:
I modified it to `""" "info"`  etc. 
It's clear that triple double quotes is the way used in the grammar to generate a double-quote, it's used both in the grammar and already in this section.
And` "info"` is similar to how strings are handled elsewhere for a literal string `info`. 
If we strictly followed the production of string it would be `""" "i" "n" "f" ...` - but that is unreadable.

Reply:
And of course the decision on #2988 is still needed.

Reply:
> It's clear that triple double quotes is the way used in the grammar to generate a double-quote, it's used both in the grammar and already in this section.

Yes, so far so good; we use double quotes to quote tokens that aren't grammar keywords.  For example:
```
"experiment"
   [ "(" [ experimentOption { "," experimentOption } ] ")" ]
```

The non-quoted names such as _experimentOption_ are production rules and will be typeset as such.

> And` "info"` is similar to how strings are handled elsewhere for a literal string `info`.

If there are examples of that, I'd consider these incorrect.

> If we strictly followed the production of string it would be `""" "i" "n" "f" ...` - but that is unreadable.

It would not only be unreasonable, it could also mean that you could have whitespace inside, which is not the intention:
```
preferredView = " i      n       f      o         "
```

Well, we actually seem to use the grammar language also for lexing definitions where white-space needs to be expressed explicitly.  For example, see 18.8.1 _Version Numbering_.  It seems like the use of upper case names in the production rules could be a hint about expressing lexing units rather than production rules, so maybe we could have done it in this extremely ugly fashion:
```
"preferredView" "=" ( INFO | … )

INFO = """ "i" "n" "f" "o" """
```

By the way, I note that the syntax highlighter for the grammar language doesn't understand that `"""` isn't an empty string followed by the opening of another string.  Fixing this is related to the current problem of expressing string literals.

Anyway, if we really insist on strings and use of the grammar language for expressing the annotation, we should have a way of properly up-quoting a string literal, which is something we might not have done before.  (The less flexible alternative of not using strings simply seems like better fit with our grammar language.)

For example – not saying we should do this, as it gets pretty messy to read – one could use single quotes around the double quotes to indicate a string literal:
- `"x"` – a single character token: not a string literal
- ''"x"' – the one character string literal `"x"`
- ''"info"' – the string literal `"info"`

As a better alternative, we could try avoiding the problem altogether by not using the grammar language at all.  This makes good sense if we don't care about non-string settings such as `preferredView = info`.  We've only done this in the context of pseudo-code records before, but maybe we could also do it without a record:
```
String preferredView;
```
The text should then state that the value must be given by a literal, and what the valid strings are.